include/itkTileMontage.hxx

```diff
@@ -681,7 +682,7 @@ TileMontage<TImageType, TCoordinate>::GenerateData()
           this->RegisterPair(referenceIndex, currentTile);
           ++m_FinishedPairs;
           // all registrations finished = 95% of total progress
-          this->UpdateProgress(m_FinishedPairs * 0.95 / m_NumberOfPairs);
+          //this->UpdateProgress(m_FinishedPairs * 0.95 / m_NumberOfPairs);
```

Comment:
Can we live without this progress update when there is only 5% left?

Reply:
This goes from 0% to 95%, and it is in this loop where the bulk of the processing time is spent.

Reply:
Ah.

Does it help to move the progress update to line 668 and 696?

Reply:
That resolves the hang!